src/python/pants/engine/process_test.py

```diff
@@ -260,11 +262,13 @@ def test_interactive_process_cannot_have_input_files_and_workspace() -> None:
         InteractiveProcess(argv=["/bin/echo"], input_digest=mock_digest, run_in_workspace=True)
 
 
-def test_find_binary_non_existent(rule_runner: RuleRunner) -> None:
+# NB: The two tests correspond to `which` being discoverable or not.
```

Comment:
How so? Is there some standard of where which is installed? It could certainly be installed on either or both.

Reply:
No standard afaict. On macOS, it's `/usr/bin`, whereas my Ubuntu devbox is both `/bin` and `/usr/bin`. This test won't test what we want if `which` is missing from both folders, but our CI should have it installed.

Reply:
But if its in both folders it doesn't test the non-which branch. It definitely would be best to setup two tmp dirs and limit the path to those dirs and symlink in the binaries you want to be available in eacb to have a controlled test.